system_hooks/src/l1_messenger.rs

```diff
@@ -264,8 +273,8 @@ pub(crate) fn send_to_l1_inner<S: EthereumLikeTypes>(
     topics.push(message_hash);
 
     system.io.emit_event(
-        ExecutionEnvironmentType::parse_ee_version_byte(caller_ee)
-            .map_err(SystemError::LeafDefect)?,
+        // Use EVM to charge gas for this operation
```

Comment:
Note: here it's becoming obvious why we need to replace this hook with EVM contract